.circleci/config.yml

```diff
@@ -145,12 +145,35 @@ jobs:
     steps:
       - checkout
       - run:
-          name: build all tools
+          name: Build all tools
           command: |
-            cd cmds
-            go install -a ./...
-            cd ../tools
-            go install -a ./...
+            (cd cmds && go install -a ./...)
+            (cd tools && go install -a ./...)
+            # Check that dead code elimination works.
+            # NB: core/installcommand, exp/builtin and exp/run disable DCE due to https://github.com/golang/go/issues/36021
+            # This is fixed in Go 1.15 so exclusions below can be removed once image uses Go 1.15 or later.
+            go version
+            go run github.com/u-root/u-root/tools/makebb \
+                -no-strip -o /tmp/bb \
+                cmds/*/* \
+                -cmds/core/installcommand -cmds/exp/builtin -cmds/exp/run \
+                pkg/uroot/test/foo
+            go tool nm /tmp/bb \
+              | grep github.com/u-root/u-root/pkg/uroot/test/bar \
+              > /tmp/bar_methods
+            if ! grep -q bar.Bar.UsedInterfaceMethod /tmp/bar_methods; then
+              echo == Expected method not found, something is wrong with the build.
+              exit 1
+            fi
+            if grep -q bar.Bar.UnusedNonInterfaceMethod /tmp/bar_methods; then
+              echo == Unused non-interface method has not been eliminated,
+              echo == dead code elimination is not working properly.
+              echo == See https://golang.org/src/cmd/link/internal/ld/deadcode.go for explanation.
+              echo == The most likely reason is use of reflect.Value.Method or .MethodByName.
+              exit 1
+            else
+              echo DCE is ok
+            fi
```

Comment:
I'm wondering if you could make this into a go test somewhere. It'll be way easier for someone making changes to run a go test somewhere than to copy & paste this into a script during local development

Reply:
so, for this i need an unstripped build of all commands, this seemed like a nice place to plug it into.
but as a go test... do we already have go tests that build bb images already?

Reply:
uroot_test.go in the main directory would probably be a good place.

Reply:
azure build failure is a red herring: it's buggy because it somehow builds new `u-root.go` against master `pkg`